### PR TITLE
refactor: remove more platform-centered files & update code

### PR DESCRIPTION
**Pull request by erisu**
### Motivation, Context & Description

Remove files and code that were specific to the platform-centered workflow that is not implemented to Electron platform.

### Testing

- `npm t`
- `cordova platform add`
- `cordova build electron`
- `cordova run electron --nobuid`

### Checklist

- [x] I've run the tests to see all new and existing tests pass
- [x] I added automated test coverage as appropriate for this change

### CODE REVIEW

**Comment by erisu** [comment]
Last review call. This PR will be merged within 8 hours.

File: lib/Api.js [hunk 1]
```diff
@@ -51,8 +51,10 @@ class Api {
     constructor (platform, platformRootDir, events) {
         this.platform = 'electron';
 
-        // MyApp/platforms/electron
-        this.root = platformRootDir || path.resolve(__dirname, '../bin/templates');
+        // The path to the platform root directory must always be provided.
+        if (!platformRootDir || !fs.existsSync(platformRootDir)) throw new CordovaError('The path to the platform root directory was undefined or invalid.');
```
Comment (brodybits) [inline lib/Api.js:55]: I think this line is too long.

File: tests/spec/unit/lib/Api.spec.js [hunk 2]
```diff
@@ -95,6 +95,15 @@ describe('Api class', () => {
             expect(api.root).toEqual(testProjectDir);
             expect(api.locations).toEqual(jasmine.objectContaining(mockExpectedLocations));
         });
+
+        it('should throw error when the constructor is missing platformRootDir argument.', () => {
+            expect(() => new Api()).toThrowError(CordovaError, 'The path to the platform root directory was undefined or invalid.');
```
Comment (brodybits) [inline tests/spec/unit/lib/Api.spec.js:100]: Prettier would reformat like this:

```suggestion
            expect(() => new Api()).toThrowError(
              CordovaError,
              "The path to the platform root directory was undefined or invalid."
            );
```

(<https://prettier.io/playground/>)

Comment (erisu) [reply tests/spec/unit/lib/Api.spec.js:100]: I applied your suggestions manually.

Also the settings you used on the prettier seemed to be all default. It uses double quotes when we use single quotes.
I went ahead made sure it was single quotes during the manual change.

I tend to keep things on one line, but i'll try to be more cautious on the length of the line.

Thanks for the review.

File: tests/spec/unit/lib/Api.spec.js [hunk 3]
```diff
@@ -95,6 +95,18 @@ describe('Api class', () => {
...
+        it('should throw error when the constructor is missing platformRootDir argument.', () => {
+            expect(() => new Api()).toThrowError(
+                CordovaError,
+                'The path to the platform root directory was undefined or invalid.'
+            );
+        });
+
+        it('should throw error when the constructor contains an invalid platformRootDir path.', () => {
+            const fakePath = path.join(__dirname, '/some-fake-path');
+            expect(() => new Api(undefined, fakePath)).toThrowError(CordovaError, 'The path to the platform root directory was undefined or invalid.');
```
Comment (brodybits) [inline tests/spec/unit/lib/Api.spec.js:108]: Can we reformat this line as well?

**Comment by brodybits** [comment]
LGTM, thanks for the edits. I took the liberty to edit the title based on the scope of the updates, feel free to change it back if it is not right.

It looks like some test cases are changed to run with the Cordova test fixture as opposed to platform-centered workflow, did not see any duplicates in the affected files.

Dumb question but I am assuming this would be merged in a squash commit.

**Comment by erisu** [comment]
Yes, this PR will be squashed.

I always squash my PRs for the most part. If I ever need to retain multiple commits I would use "Rebase & commit" but then try to manually squash commit that are not needed in git history. E.g. `chore: apply suggestion`.

**Comment by erisu** [comment]
Also to point out the commit message of the squashed PR will be this PRs title. 

`refactor: remove more platform-centered files & update code`

You change to the title is OK. Some times my titles are short because I trying to meet the git commit first line character length limitation. This title might be too long but it will just ellipse the rest to the next body of the commit message.

**Comment by brodybits** [comment]
> Some times my titles are short because I am trying to meet the git commit first line character length limitation.

Yup, I generally try to do the same thing with GitHub.

My trick is to leave off the prefix word, like "chore" or "refactor", to get within the "standard" 50 characters, which of course breaks the "conventional commit" standard that more and more people are using. Leaving it up to you:)

**Pull request merged**